pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -152,32 +152,34 @@ def update_next_infection_status(self, person: Person):
                                        InfectionStatus.Dead,
                                        InfectionStatus.Vaccinated]:
             person.next_infection_status = None
+            return
         elif (person.care_home_resident and
               person.infection_status == InfectionStatus.InfectICU):
             person.next_infection_status = InfectionStatus.Dead
+            return
         elif (person.care_home_resident and
               person.infection_status == InfectionStatus.InfectHosp):
             carehome_params = Parameters.instance().carehome_params
             carehome_hosp = carehome_params['carehome_rel_prob_hosp']
             if random.uniform(0, 1) > carehome_hosp:
```

Comment:
ERROR 3: This error caused the simulation to crash when it satisfied the conditions of a person being a care home resident and also having InfectHosp status. The reason for it was that if the above random.uniform test is not greater than carehome_hosp, then the person does not have their next_infection_status changed. This meant that the simulation had some people with infection_status and next_infection_status set to InfectHosp, causing an error. We fixed this by changing the logic of the statements so that these people would have their next_infection_status changed.

Reply:
Again, really great work here - I'm very impressed by this, and feeling guilty we didn't pick these up before